File: 86/Solution.java
class Solution {
    public class ListNode {
        int val;
        ListNode next;
        ListNode() {}
        ListNode(int val) { this.val = val; }
        ListNode(int val, ListNode next) { this.val = val; this.next = next; }
    }

    public ListNode partition(ListNode head, int x) {
        if (head == null) {
            return head;
        }
        ListNode small = new ListNode(0, null);
        ListNode small_head = small;
        ListNode big = new ListNode(0, null);
        ListNode big_head = big;
        while (head != null) {
            if (head.val < x) {
                small.next = head;
                small = small.next;
            } else {
                big.next = head;
                big = big.next;
            }
            head = head.next;
        }
        small.next = big_head.next;
        big.next = null;
        return small_head.next;
    }
}

File: 86/solution.go
package _86

type ListNode struct {
	Val  int
	Next *ListNode
}

func partition(head *ListNode, x int) *ListNode {
	if head == nil {
		return head
	}
	big := &ListNode{0, nil}
	big_head := big
	small := &ListNode{0, nil}
	small_head := small
	for head != nil {
		if head.Val < x {
			small.Next = head
			small = small.Next
		} else {
			big.Next = head
			big = big.Next
		}
		head = head.Next
	}
	small.Next = big_head.Next
	big.Next = nil
	return small_head.Next
}
